src/snowflake/snowpark/modin/plugin/_internal/indexing_utils.py

```diff
@@ -2213,6 +2216,36 @@ def set_frame_2d_labels(
     index_is_frame = isinstance(index, InternalFrame)
     item_is_frame = isinstance(item, InternalFrame)
     item_is_scalar = is_scalar(item)
+    original_index = index
+    # If `item` is from a Series (rather than a Dataframe), flip the series item values to apply them
+    # across columns rather than rows.
+    if frame_is_df_and_item_is_series and (columns == slice(None) or len(columns) > 1):  # type: ignore[arg-type]
```

Comment:
This should be done in `_set_2d_labels_helper_for_frame_item`

Reply:
I actually think it needs to be done in this method, since we need to modify `item` before the map is created (which is passed into `_set_2d_labels_helper_for_frame_item`, and we need the modified item later on in this method.

Reply:
I can move this into the conditional for `if item_is_frame` though!